2_process/src/munge_GFv1_catchments.R

```diff
@@ -0,0 +1,146 @@
+  #' prms_lines must contain variables subsegid and geometry
+  #' @param segs_w_comids data frame containing the PRMS segment ids and the contributing COMID's
+  #' segs_w_comids must contain variables PRMS_segid and COMID
+  #' @param prms_hrus sf (multi)polygon containing the HRU's from the NHGFv1
+  #' @param segs_split character vector indicating the PRMS_segid of the segments that were split in 
+  #' the delaware-model-prep pipeline
+  #' @param crs_out numeric EPSG code indicating desired crs. Defaults to EPSG: 4326, WGS84.
+  #
+  
+  # 1. Fill PRMS lines data frame so that every segment has a value for the attribute
+  # `hru_segment`, which can be used as a key to join each segment to its corresponding HRU polygons.
+  prms_hrus_filled <- prms_lines %>%
+    # Transform prms_lines into a list with a data frame for each PRMS segment (represented by a unique subsegid) 
+    split(.,.$subsegid) %>%
+    # For each PRMS segment, find the HRU with the greatest overlap and add as an attribute
+    lapply(.,function(x){
+      x$hru_segment <- ifelse(x$subsegseg %in% prms_hrus$hru_segment,
+                              x$subsegseg,
+                              find_intersecting_hru(x,prms_hrus))
+      return(x)
+    }) %>%
+    # Reformat list into a data frame with one row per PRMS segment
+    bind_rows()
+  
+  # 2. For select segments, redefine the catchment polygon using the contributing NHDv2 catchments
+  prms_splitsegs <- prms_hrus_filled %>%
+    # Filter PRMS segments to only include select segments that require special handling
+    filter(subsegid %in% segs_split) %>%
+    # Transform PRMS segments into a list with a data frame for each segment
+    split(.,.$subsegid) %>%
+    # For each segment requiring special handling, return the polygon that represents the catchment area
+    lapply(.,return_nhdv2_cat_shps, segs_w_comids = segs_w_comids) %>%
+    # Reformat list into a data frame with one row per PRMS segment
+    bind_rows()
+  
+  # 3. Concatenate munged catchment polygons
+  catchments_ls <- list()
+  
+  # Fill empty list with appropriate catchments
+  for(i in seq_along(prms_hrus_filled$subsegid)){
+    
+    prms_line <- prms_hrus_filled[i,]
+    
+    # If PRMS segment is one of the segments that require special handling, 
+    # retrieve NHDPlusV2 catchments and use those as the catchment boundary.
+    # Else, represent the catchment boundaries using intersecting HRU polygon.
+    if(prms_line$subsegid %in% segs_split){
+      cat <- prms_splitsegs %>%
+        filter(PRMS_segid == prms_line$subsegid) %>%
+        mutate(hru_segment = prms_line$subsegid,
+               hru_id = NA) %>%
+        select(PRMS_segid,hru_segment,hru_id,geometry) %>%
+        sf::st_transform(.,crs_out)
+    } else {
+      cat <- prms_hrus %>%
+        filter(hru_segment == prms_line$hru_segment) %>%
+        mutate(PRMS_segid = prms_line$subsegid) %>%
+        select(PRMS_segid,hru_segment,hru_id) %>%
```

Comment:
same comment as above.

Reply:
It looks like the attribute `hru_area` from NHGFv1 is area given in acres (see metadata [here](https://www.sciencebase.gov/catalog/file/get/535eda80e4b08e65d60fc834?f=__disk__12%2F3e%2F81%2F123e81778296aec8dc694697f4e106583aa21e6b&transform=1&allowOpen=true)). So instead I kept the column named `Shape_Area` and renamed it `hru_area_m2` to match the code chunk above. Would you agree with that interpretation?

Reply:
ahh good find that is suprising. How large are the values once in m2? Could it be worth putting into square km?

Reply:
I left in m2 to be consistent with `Shape_Area` (and to an extent, to also avoid propagating rounding errors). If you'd prefer km2 I'd be fine with that though.